net/tstun/tun_features_linux.go

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package tstun
+
+import (
+	"os"
+	"strings"
+
+	"github.com/tailscale/wireguard-go/tun"
+)
+
+func setLinkFeatures(dev tun.Device) error {
+	val := os.Getenv("TS_TUN_DISABLE_UDP_GRO")
+	if strings.ToLower(val) == "true" || val == "1" {
```

Comment:
Minor nit: we could perhaps either perform the strings.ToLower(...) then a named variable like "disable" and perform the disabling or invert and return early so:

```go
val := strings.ToLower(os.Getenv(...))
disable := val == "true" || val == "1"
if disable {
  linuxDev, ok := dev.(tun.LinuxDevice)
  if ok {
    linuxDev.DisableUDPGPRO()
  }
}

if
```

OR

```go
val := strings.ToLower(os.Getenv(...))
disable := val == "true" || val == "1"
if !disable {
   return nil
}

if linuxDev, ok := dev.(tun.LinuxDevice); ok {
    linuxDev.DisableUDPGPRO()
}
return nil
```